hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -193,6 +196,32 @@ public TransactionContext startTransaction(
     return handleStartTransactionRequests(raftClientRequest, omRequest);
   }
 
+  @Override
+  public TransactionContext preAppendTransaction(TransactionContext trx)
```

Comment:
Can we add a unit test for this method since it has a few different execution flows?

Reply:
Yes, done.